benchmark.gradle

```diff
@@ -0,0 +1,90 @@
+sourceSets {
+	benchmark {
+		java.srcDir 'src/benchmark/java'
+		resources.srcDir 'src/benchmark/resources'
+		compileClasspath += main.output
+		runtimeClasspath += main.output
+	}
+}
+
+def benchmarksAvailable = sourceSets.benchmark.java.getFiles().any { f -> f.exists() }
+
+dependencies {
+	benchmarkCompile 'org.openjdk.jmh:jmh-core:1.22'
+	benchmarkAnnotationProcessor 'org.openjdk.jmh:jmh-generator-annprocess:1.22'
+}
+
+configurations {
+	benchmarkCompile {
+		extendsFrom compile
+	}
+
+	benchmarkImplementation {
+		extendsFrom implementation
+		extendsFrom benchmarkCompile
+	}
+
+	benchmark {
+		extendsFrom runtime
+		extendsFrom benchmarkCompile
+	}
+}
+
+compileBenchmarkJava {
+	enabled benchmarksAvailable
+	options.compilerArgs = compileJava.options.compilerArgs
+	options.fork = compileJava.options.fork
+	options.forkOptions = compileJava.options.forkOptions
+}
+
+task benchmarkRun(type: JavaExec) {
+	enabled benchmarksAvailable
+	description 'Run benchmarks. Use property benchmarkFilter to set benchmark filtering regex.'
+	group 'Verification'
+	classpath = sourceSets.benchmark.runtimeClasspath
```

Comment:
I think jmh works by default with jmh directory not benchmark, am I wrong?

Reply:
Not sure there is a 'default' directory. I opted for benchmark in similar fashion to 'test' which describes the purpose of the code in the sourceset, but not the test framework (junit, testng, ...).

No problem to change if there is a preference for jmh.

Working on getting the CLA signed.

Reply:
If `jmh` is the default, then that's the way I would go, as well.